syntax = "proto3";

package ecron;


enum ErrorCode {
    ENOERROR                = 0;     // No Error
    ENOSERVICE              = 1001;  // Service not found
    ENOMETHOD               = 1002;  // Method not found
    EREQUEST                = 1003;  // Bad Request
    ERESPONSE               = 1004;  // Bad Response
    EINTERNAL               = 1005;  // Internal Server Error
}


message RpcMeta {
    RpcRequestMeta request = 1;
    RpcResponseMeta response = 2;
    int64 correlation_id = 4; 
}

message RpcRequestMeta {
    string service_name = 1;
    string method_name = 2;
}

message RpcResponseMeta {
    int32 error_code = 1;
    string error_text = 2;
}
